test/distributed/fsdp/test_fsdp_mixed_precision.py

```diff
@@ -761,41 +766,54 @@ def test_eval_root_cast_inputs(self):
         """
         In a case where root module does not manage FSDP parameters,
         ensure that we don't cast forward inputs which could potentially
-        cause a dtype mismatch.
+        cause a dtype mismatch. Check that FSDP_FULL_PREC_IN_EVAL controls
+        this.
         """
 
+        reduce_dtype = torch.float16
+
         class MyModel(torch.nn.Module):
             def __init__(self):
                 super().__init__()
                 self.a = nn.Linear(5, 5)
 
-            def forward(self, x):
-                assert x.dtype == torch.float32, f"Expected fp32, got {x.dtype}"
+            def forward(self, x, expect_full_prec_in_eval):
+                if expect_full_prec_in_eval:
+                    assert x.dtype == torch.float32, f"Expected fp32, got {x.dtype}"
+                else:
+                    assert x.dtype == reduce_dtype, f"Expected {reduce_dtype}, got {x.dtype}"
```

Comment:
I guess the pain here is that t he `self` arg to `forward()` shadows the test class `self`, so it is hard to use `self.assertEquals()`.